src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs

```diff
@@ -250,12 +250,6 @@ private void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context, INamedTypeSymb
             // In this case, inlining the 'i' would cause it to longer be definitely
             // assigned in the WriteLine invocation.
 
-            if (localDeclarator.Initializer == null)
```

Comment:
:memo: I'm concerned about the performance implications of this change. I find that it's a clear usability win and the test case seems valid, but the solution results in a great deal of additional analysis. I'm not sure what the best approach to take here is.